library/net.cpp

```diff
@@ -792,29 +784,31 @@ static void dnet_process_socket(dnet_connect_state &state, epoll_event &ev)
 			break;
 		}
 
-		dnet_log(state.node, DNET_LOG_NOTICE, "%s: received indexes shard count: local: %d, remote: %d, using server one",
-				dnet_addr_string(&socket->addr), state.node->indexes_shard_count, indexes_shard_count);
+		dnet_log(state->node, DNET_LOG_NOTICE, "%s: received indexes shard count: local: %d, remote: %d, using server one",
+				dnet_addr_string(&socket->addr), state->node->indexes_shard_count, indexes_shard_count);
 
-		if (indexes_shard_count != state.node->indexes_shard_count && indexes_shard_count != 0) {
-			dnet_log(state.node, DNET_LOG_INFO, "%s: local and remote indexes shard count are different: "
+		if (indexes_shard_count != state->node->indexes_shard_count && indexes_shard_count != 0) {
+			dnet_log(state->node, DNET_LOG_INFO, "%s: local and remote indexes shard count are different: "
 					"local: %d, remote: %d, using remote (%d) one",
 					dnet_addr_string(&socket->addr),
-					state.node->indexes_shard_count, indexes_shard_count, indexes_shard_count);
+					state->node->indexes_shard_count, indexes_shard_count, indexes_shard_count);
 
-			state.node->indexes_shard_count = indexes_shard_count;
+			state->node->indexes_shard_count = indexes_shard_count;
 		}
 
-		socket->buffer = malloc(cmd->size);
-		socket->io_data = socket->buffer;
-		socket->io_size = cmd->size;
-
-		if (!socket->buffer) {
+		char *buffer = new(std::nothrow) char[cmd->size];
```

Comment:
I think it is not necessary here, because buffer is fixed-size and it doesn't need any extra vector semantics.